src/filters/common.js

```diff
@@ -0,0 +1,137 @@
+/**
+ * @license Copyright (c) 2003-2018, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+import HtmlDataProcessor from '@ckeditor/ckeditor5-engine/src/dataprocessor/htmldataprocessor';
+
+const htmlDataProcessor = new HtmlDataProcessor();
+
+/**
+ * Extracts `body` tag contents from the provided HTML string.
+ *
+ * @param {Object} data
```

Comment:
I don't like how these functions are designed to work. From the point of view of these utils there's very little sense in working on some arbitrary `data` object. E.g. this specific function accepts an HTML string and returns another HTML string. Period. The `data` object is a noise here. Plus, modifying an object that was passed to a function isn't usually a good practice.

Imagine, for instance, than in a month's time you'll need to use it somewhere else to process some HTML. This time, not together with the `transformInput()` function, but alone. You will either have to change this function or use it with some object created for that call.

Reply:
True, I had some doubts about it too TBH. Makes perfect sense to refactor them in a way you mentioned 👍